unlock-app/src/components/developer/DeveloperOverlay.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import NoSSR from 'react-no-ssr'
+import styled from 'styled-components'
+
+import UnlockPropTypes from '../../propTypes'
+import { setProvider } from '../../actions/provider'
+import withConfig from '../../utils/withConfig'
+
+export function DeveloperOverlay({ config, selected, setProvider }) {
+  const providers = Object.keys(config.providers)
+  if (config.env !== 'dev') return null
+
+  return (
+    <NoSSR>
+      <Overlay>
+        <Inner>
+          <select
+            value={selected}
+            name="changeProvider"
+            id="changeProvider"
+            onChange={e => setProvider(e.target.value)}
+          >
+            {providers.map(provider => (
+              <option value={provider} key={provider}>
+                {provider}
+              </option>
+            ))}
+          </select>
+          <div>Choose Provider</div>
+        </Inner>
+      </Overlay>
+    </NoSSR>
+  )
+}
+
+DeveloperOverlay.propTypes = {
+  config: UnlockPropTypes.configuration.isRequired,
+  selected: UnlockPropTypes.provider,
+  setProvider: UnlockPropTypes.callback.isRequired,
```

Comment:
I'd argue that this is actually not even a `callback` in the JS sense, but really a regular function invoked on user actions ;) Let's just stick to `PropTypes.func`!